### PR TITLE
Add scspell in C.I

### PR DESCRIPTION
**Pull request by BibianaC**
This a first PR. I had a few questions.

- I have added @robredpath dictionary and added a few other words. I find it difficult to know where to add a word that is not a python nor a natural language word. For example two words that we put together in a function name.
- Do we want to check the docs folder or only the flattentool one?
- Do we want to check only `.py` docs or all?

### CODE REVIEW

**Comment by BibianaC** [comment]
[Use scspell in C.I.](https://trello.com/c/vTtZJ3Lw/47-use-scspell-in-ci)

**Comment by odscjames** [comment]
> Do we want to check the docs folder or only the flattentool one?
> Do we want to check only .py docs or all?

Personal thought; given this is a trial I'm happy with us starting by checking a limited set of things only and we'll see how it goes

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -22,3 +22,9 @@ jobs:
     - run: black --check *.py */
     - run: isort --check-only *.py */
     - run: flake8
+    - run: pip install scspell3k
```
Comment (odscjames) [inline .github/workflows/lint.yml:25]: This line shouldn't be needed as it's in req_dev and that is installed a few lines up?

**Comment by odscjames** [comment]
1)

```
- run: scspell flattentool/*.py
- run: scspell flattentool/tests/*.py

```

This bothered me - it would be better to use a recursive mode if possible here. The problem with this is that if someone adds a new directory eg `flattentool/lib/process.py` then that file won't be automatically checked without adding a new line to the CI file - something it would be very easy to miss.

But it seems not to have a recursive mode? Odd. If it really doesn't we'll just have to live with this.

**Comment by odscjames** [comment]
2)

I ran it and got a hang at a prompt:
```

(.ve) odscjames@odscjames-Apollo:~/work/flatten-tool$ scspell flattentool/*.py
flattentool/cli.py:3: Unmatched 'argparse' --> {argparse}
   (i)gnore, (I)gnore all, (r)eplace, (R)eplace all, (a)dd to dictionary, or
   show (c)ontext? [i]

```

This won't work on a C.I. server - the C.I. server will just keep running forever and then they will think we are mining B**coin and shut us down :-)

I see there is a --report-only mode - this may be what we want?

For a C.I. server, what we need is a command that returns exit code 0 if all fine and exit code something else if there is a problem - can you see if --report-only does that and add it if so?

(For C.I. work, I like to take a new branch of this branch and then add a commit that deliberately breaks something - then see if that C.I. fails! Links can  then be added to the broken C.I. run for a reviewer to see)

**Comment by odscjames** [comment]
3)

This seems to put everything into one directory; I'm not sure this is the best approach when scspell lets you provide multiple dictionaries. Did we look into how well that works and if we can do that?

If we do stick with this approach, we at least need to note where the original directory came from and what license it was under.

**Comment by robredpath** [comment]
> This bothered me - it would be better to use a recursive mode if possible here.

Maybe use `find` and `xargs` to construct something here?

**Comment by BibianaC** [comment]
> 
> I see there is a --report-only mode - this may be what we want?
> 

I have ran this locally but I'm not sure it would make the CI exit. Something to check.

**Comment by BibianaC** [comment]
> This seems to put everything into one dictionary; I'm not sure this is the best approach when scspell lets you provide multiple dictionaries. Did we look into how well that works and if we can do that?
> 
> If we do stick with this approach, we at least need to note where the original dictionary came from and what license it was under.

In the file there is more than one dictionary. 

I have not found how to add more than just the language or natural one. When there is a spelling mistake and you choose `add to a dictionary` then the only possibilities are `natural language` and the `programming language` chosen.

There has to be a way to add more but the documentation is not that clear about it. Maybe something that can be done manually instead of using the terminal.

**Comment by BibianaC** [comment]
> This bothered me - it would be better to use a recursive mode if possible here. 

Maybe would be possible to add this to a file and then ask to run that file?

**Comment by odscjames** [comment]
Ok, and I've been trying this and I'm actually pretty unimpressed.

* It reported both Europe and London as wrong, despite them being in the base directory
* It reported kwargs_from_parsed_args as wrong, despite the readme claiming it splits by underscores

I'll push up a pull request with what I've got so far, but I'm going to think about this and maybe do some googling.

**Comment by odscjames** [comment]
https://github.com/OpenDataServices/flatten-tool/pull/379

But it does work recursively, use the report flag, have a British dictionary and has a checked in dictionary with only our special words, so the file is of a decent size and we get upstream changes.

**Comment by odscjames** [comment]
Closing after discussion in today's dev best practices meeting.

**Pull request closed without merging**